### ISSUE TITLE
scrolling with two-finger gesture doesn't work

### ISSUE THREAD

**Issue by wtholliday** [MacDownApp/macdown#203]
I'm using OS X 10.10 on a rMBP. MacDown 0.3 (485)
1. Open the MacDown Help from the help menu.
2. Two-finger scroll the output view on the right
3. observe: relatively smooth scrolling performance (still not as good as Safari)
4. Two-finger scroll the markdown view on the left
5. observe: choppy scrolling performance with jumps the wrong way which makes it very hard to scroll through the entire document

**Comment by dizballanze** [comment]
+1

**Comment by mbosner** [comment]
@uranusjr
The problem is gone with enjamintanweihaos hint.

**Comment by ronen** [comment]
I'm in the same boat as @Lhfcws in #271 -- the @benjamintanweihao hint doesn't help me.  (Unless you need to reboot or something afterwards?)

I do have a MPB with Retina, but I haven't experienced any other programs with the same problem.

**Comment by uranusjr** [comment]
I really want to fix this too, but it’s hard to locate the problem without a Retina Mac. If anyone is willing to help, please drop a word. I need to build MacDown from source to debug, so you’ll need to set up a dev environment (Xcode and CocoaPods), and at least know how to locate files, type/delete things, and build things in Xcode. You *don’t* need to know how to write Objective-C—I’ll provide the code—but it would be would be awesome if you know something about it.

**Comment by JanX2** [comment]
@uranusjr You can set things up to be retina on your machine. This may get cramped though.

<http://stackoverflow.com/questions/12124576/how-to-simulate-a-retina-display-hidpi-mode-in-mac-os-x-10-8-mountain-lion-on>

**Comment by uranusjr** [comment]
@JanX2 I tried that, but still couldn’t reproduce the behaviour. I suspect this is not happening on all Retina machines, but only with some specific combinations of hardware and configuration. Anyway, that workaround does not work. :(

**Comment by JanX2** [comment]
@uranusjr Does the markdown NSTextView have “Non-contiguous Layout” enabled? This can often cause issues with fancy text views.

**Comment by uranusjr** [comment]
Nope. (It was enabled in early versions, but was advised to fix another issue.)

**Comment by saurabhg17** [comment]
I have same problem: choppy scrolling performance in the editor view. It doesn't matter whether I use mouse or touch pad. Scrolling in the preview pane is fine but it doesn't sync with editor view which I suspect is because of scrolling problems.

This makes MacDown pretty much unusable as of now. I can try to help find the issue. I am a windows developer and new to OS X so any help or pointers are appreciated.

**Comment by uranusjr** [comment]
With help from @saurabhg17 I’m able to come up with a possible fix for this problem. Download the [latest testing build](http://macdown.uranusjr.com/download/v0.4.3d13/) and try it out. If you do, please report whether this problem is fixed or not.

**Comment by ronen** [comment]
>  Download the latest testing build and try it out.

Thanks for working on this!   But unfortunately still didn't fix the problem for me :cry:

BTW I find that it sometimes I can have two files open, and one will scroll perfectly smoothly while the other has the scrolling problem.  I haven't yet figured out how to get this reproducible.  If I do I'll be sure to let you know.

**Comment by saurabhg17** [comment]
If latest build still doesn’t work for you. Can you try this:
Open a file, test if scrolling doesn’t work, then hide the preview pane. Now drag the preview pane back and see if this fixes scrolling.

Regards,
Saurabh

**Comment by dizballanze** [comment]
It is doesn't work for me. It only works without preview panel.

**Comment by ronen** [comment]
@saurabhg17, here's my latest experience:

When I first opened the file, scrolling was fine.  I opened another file in another window, it was fine too.  Randomly worked with the files a little while, eventually scrolling stopped working in one of the windows; I still don't know what triggered it.

In the window with broken scrolling, with the preview pane hidden scrolling is fine.  But when I restore the preview pane (via shift-cmd-H, is that what you meant by 'drag back'), the scrolling is broken again.

**Comment by uranusjr** [comment]
I’ve made a custom build, available [here](http://d.pr/f/1iDvV). Could you check if this works? This could help clarifying some things.

(Technical note: This version removes the width adjustment code inside `-boundsDidChange:` completely.)

**Comment by dizballanze** [comment]
@uranusjr It works fine for me!

**Comment by ronen** [comment]
@uranusjr it works for me too.  Thanks!
